### ISSUE TITLE
error: Cannot read property 'path' of undefined

### ISSUE THREAD

**Issue by TanninOne** [agc93/beatvortex#4]
Please see this issue: https://github.com/Nexus-Mods/Vortex/issues/6710

The problem is in this line (compiled): var gamePath = this._api.getState().settings.gameMode.discovered[__1.GAME_ID].path;

This is a fairly common mistake that happens to us as well: The "discovered" object stores manually configured settings for a game (e.g. if you manually set up tools). These settings don't get removed if a game is no longer discovered but the path gets set to undefined.
So if a game was once installed, got configured, then uninstalled you may have an entry for the game in discovered but the path is undefined.

The line further down you use that gamePath in path.join, which doesn't like undefined as a parameter.

**Comment by agc93** [comment]
I've removed the two places that should've caused this to happen on quite the scale it did, sorry about that :( [v0.3.5](https://github.com/agc93/beatvortex/releases/tag/v0.3.5) should be a bit safer.

There's still a lot of places using that same method for getting the install path so I will need to progressively add null-checking to them as I go through.

**Comment by agc93** [comment]
So I'm understanding this properly @TanninOne I should be able to replicate this by 'unmanaging' my game (i.e. delete all its profiles) and then trigger a deploy (or whatever starts the problematic code)?

**Comment by TanninOne** [comment]
No, afaict the only way to reproduce is by manually configuring something for the game (e.g. a tool on the dashboard) and then uninstalling the game (from steam).
That way the next time Vortex starts it will no longer have the game path set.

**Comment by agc93** [comment]
Alright thanks, I'll break out the Hyper-V and see if I can replicate it just so I can check any other spots I might have done this
